server/chat-plugins/chatlog.ts

```diff
@@ -380,4 +601,26 @@ export const commands: ChatCommands = {
 		const roomid = targetRoom ? targetRoom.roomid : target;
 		this.parse(`/join view-chatlog-${roomid}--today`);
 	},
+
+	sl: 'searchlogs',
+	searchlog: 'searchlogs',
+	searchlogs(target, room, user, connection, cmd) {
+		target = target.trim();
+		const [search, tarRoom, cap, date] = target.split(',') as [string, string, number, number];
+		if (!target) return this.parse('/help searchlogs');
+		if (!search) return this.errorReply('Specify a query to search the logs for.');
+		if (cap && isNaN(cap) && toID(cap) !== 'all') return this.errorReply(`Cap must be a number or [all].`);
+		const input = search.includes('|') ? search.split('|').map(item => item.trim()).join('-') : search;
+		const currentMonth = Chat.toTimestamp(new Date()).split(' ')[0].slice(0, -3);
+		const curRoom = tarRoom ? Rooms.search(tarRoom) : room;
+		const limit = cap ? `--${cap}` : `--500`;
+		return this.parse(`/join view-chatlog-${curRoom}--${date ? date : currentMonth}--search-${input}${limit}`);
```

Comment:
This still has the problem I mentioned a long time about not supporting searching for things that contain things other than numbers or lowercase letters.

Reply:
Modlog uses Dashycode to solve this problem; I continue to recommend it here.

Reply:
My problem with dashycode is that it can create ids with '--', which breaks the chatlog opts.
Not really sure what to do about that.

Reply:
I guess push it to '---' as the delimiter? That seems rather ridiculous, though.

Reply:
Yeah, I can understand how that would be a problem. You can use `splitFirst` (I think it's available as `Chat.splitFirst`) which partially solves it, and then manually use a regex to see if it ends with `--page-[number]`.

Reply:
I know, it's ugly, but I can't think of a better solution, and I think it's worth it.